test/functional/test_framework/socks5.py

```diff
@@ -117,6 +135,38 @@ def handle(self):
             cmdin = Socks5Command(cmd, atyp, addr, port, username, password)
             self.serv.queue.put(cmdin)
             logger.debug('Proxy: %s', cmdin)
+
+            requested_to_addr = addr.decode("utf-8")
+            requested_to = format_addr_port(requested_to_addr, port)
+
+            if self.serv.conf.destinations_factory is not None:
+                dest = self.serv.conf.destinations_factory(requested_to_addr, port)
+                if dest is not None:
+                    logger.debug(f"Serving connection to {requested_to}, will redirect it to "
+                                 f"{dest['actual_to_addr']}:{dest['actual_to_port']} instead")
+                    with socket.create_connection((dest["actual_to_addr"], dest["actual_to_port"])) as conn_to:
+                        self.conn.setblocking(False)
+                        conn_to.setblocking(False)
+                        sockets = [self.conn, conn_to]
+                        done = False
+                        while not done:
```

Comment:
Maybe factor this shovel-from-socketa-to-socketb loop out to a function in netutil as well.

Reply:
I moved it to its own function, but in `test/functional/test_framework/socks5.py` because it uses the newly created `sendall()` (due to the [suggestion below](https://github.com/bitcoin/bitcoin/pull/29420#discussion_r1802683914)) which is very similar to `recvall()` which already exists in `socks5.py`.

I guess if this is to be moved to `test/functional/test_framework/netutil.py`, then all 3 of `recvall()`, `sendall()` and `forward_sockets()` should be moved. Leave it as is or move to `netutil.py`?

Reply:
thanks!
factoring it out was more fo a readablity thing, i don't really mind if it's in netutil
i'm fine with leaving it where it is, if some other code needs it it can always be moved to a more general place